### DISCUSSION TITLE
How to set log level for specific packages using log4j2.xml and @Logging annotation?

### DISCUSSION THREAD

**Discussion by mcmessina** [aws-powertools/powertools-lambda-java#1730] [Q&A]
I would like to set the log level for a specific package, such as the following (I added one line to the default example provided in the documentation [here](https://docs.powertools.aws.dev/lambda/java/core/logging/#__tabbed_2_1) and set the root log level to "TRACE"):
```xml
<?xml version="1.0" encoding="UTF-8"?>
<Configuration>
    <Appenders>
        <Console name="JsonAppender" target="SYSTEM_OUT">
            <JsonTemplateLayout eventTemplateUri="classpath:LambdaJsonLayout.json" />
        </Console>
    </Appenders>
    <Loggers>
        <!--Set a new log level for a specific package here-->
        <Logger name="software.amazon" level="WARN"/>
        <Logger name="JsonLogger" level="TRACE" additivity="false">
            <AppenderRef ref="JsonAppender"/>
        </Logger>
        <Root level="trace">
            <AppenderRef ref="JsonAppender"/>
        </Root>
    </Loggers>
</Configuration>
```
Is this type of configuration modification supported? When I try this with the `@Logging` annotation in a Java 11-based Lambda, these level-based statements are ignored (but working locally). Could this be related to the the way I package the application using Maven? This only seems to work if I modify the logging configuration within the Java handler like the following:
```java
Configurator.setLevel("software.amazon", Level.WARN);
```

**Comment by phipag** [comment]
Hi @mcmessina, I am currently reviewing some older discussion topics and see that this is almost a year old. Please let me know if this issue still persists. I am resolving the discussion for now but please feel free to re-open and we can debug this together. Note that we have also released v2 of this library.
